OOModel/src/declarations/Class.cpp

```diff
@@ -207,6 +207,17 @@ QSet<Class*> Class::allBaseClasses()
 	return bases;
 }
 
+QSet<Class*> Class::directBaseClasses()
+{
+	QSet<Class*> result;
+	for (auto baseClass : *baseClasses())
+		if (auto asClass = expressionToClass(baseClass))
+			result << asClass;
+	if (implicitBaseFromProject())
```

Comment:
I think you should only use the implicit base class in case there are no other direct base classes. E.g. right now you'll alway add `Object` as a base class, but it's not always a direct base class.